kernel/src/net/socket/ip/options.rs

```diff
@@ -0,0 +1,32 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use crate::impl_socket_options;
+
+/// IP-level socket options.
+#[derive(Debug, Clone, Copy, CopyGetters, Setters)]
+#[get_copy = "pub"]
+#[set = "pub"]
+pub(super) struct IpOptionSet {
+    tos: u8,
```

Comment:
Should we use `Option<NonZeroU8>` for `tos` since `tos` is either `-1` or in `[1, 255]`?